library/extractor/src/main/java/com/google/android/exoplayer2/extractor/ts/H265Reader.java

```diff
@@ -243,10 +243,20 @@ private static Format parseMediaFormat(
     bitArray.skipBits(40 + 4); // NAL header, sps_video_parameter_set_id
     int maxSubLayersMinus1 = bitArray.readBits(3);
     bitArray.skipBit(); // sps_temporal_id_nesting_flag
-
-    // profile_tier_level(1, sps_max_sub_layers_minus1)
-    bitArray.skipBits(88); // if (profilePresentFlag) {...}
-    bitArray.skipBits(8); // general_level_idc
+    int generalProfileSpace = bitArray.readBits(2);
+    boolean generalTierFlag = bitArray.readBit();
+    int generalProfileIdc = bitArray.readBits(5);
+    int generalProfileCompatibilityFlags = 0;
+    for (int i = 0; i < 32; i++) {
```

Comment:
Is this for loop the same as `int generalProfileCompatibilityFlags = bitArray.readBits(32)`?

[I don't think there's any reversing going on, but if there is you could probably use `Integer.reverse` to get them the right way round]

Reply:
This code is just moved from [`buildHevcCodecStringFromSps`](https://github.com/google/ExoPlayer/pull/9421/files/45db77dc6ddfb0ca9de191faad184aaf90821012#diff-bfa623b40c5dce28bfd03d9cc30f13aca7fdbbbe035158284d12e2e61a2b31b3L101) I didn't take the risk of changing any logic here.

But it should be not equal to `bitArray.readBits(32)` because Java is big endian and the for loop reverses the bit order. `Integer.reverse` would probably work.

But as mentioned, I didn't want to cleanup/deduplicate existing code here. And when changing it in this place, I should problably also change it in `NalUnitUtil.java`. Do you want that?